src/rootscripts/plotGui.C

```diff
@@ -318,150 +337,199 @@ void plotGuiMainFrame::PrintSelected(const int sel1, const int sel2)
+		g2->SetName (graphName.str().c_str());
+		g2->SetTitle(graphName.str().c_str());
+		g2->SetMarkerStyle(21);
+		g2->SetMarkerSize(0.5);
+		g2->SetMarkerColor(_usefulColors[treeIndex % _usefulColors.size()]);
+		g2->SetLineColor(_usefulColors[treeIndex % _usefulColors.size()]);
+
+		for(unsigned int i = 0; i < nPoints; ++i){
+			const fitResult* result = _fitResults[treeIndex][i];
+
+			if(not result->converged()) {
+				continue;
+			}
 
-		if((numeric_limits<double>::has_infinity and intensity1 == numeric_limits<double>::infinity()) or intensity1!=intensity1) {
-			continue;
-		}
+			int wi1 = result->waveIndex(w1);
+			int wi2 = result->waveIndex(w2);
+			double intensity1 = 0.;
+			double intensity2 = 0.;
+			double intensityErr1 = 0.;
+			double intensityErr2 = 0.;
+			double ph = 0.;
+			double phErr = 0.;
+			complex<double> rho(0., 0.);
+			TMatrixT<double> rhoCov(2, 2);
+			if(wi1 >= 0) {
+				intensity1 = result->intensity(w1.c_str());
+				intensityErr1 = result->intensityErr(w1.c_str());
+			}
+			if(wi2 >= 0) {
+				intensity2 = result->intensity(w2.c_str());
+				intensityErr2 = result->intensityErr(w2.c_str());
+			}
+			if(wi1 >= 0 and wi2 >= 0) {
+				ph = result->phase(w1.c_str(), w2.c_str());
+				phErr = result->phaseErr(w1.c_str(), w2.c_str());
+				rho = result->spinDensityMatrixElem(wi1, wi2);
+				rhoCov = result->spinDensityMatrixElemCov(wi1, wi2);
+			}
 
-		g1->SetPoint(i, result->massBinCenter()*0.001, intensity1);
-		g1->SetPointError(i, _binWidth*0.5, intensityErr1);
+			if((numeric_limits<double>::has_infinity and intensity1 == numeric_limits<double>::infinity()) or intensity1!=intensity1) {
+				continue;
+			}
 
-		if((numeric_limits<double>::has_infinity and intensity2 == numeric_limits<double>::infinity()) or intensity2!=intensity2) {
-			continue;
-		}
+			g1->SetPoint(i, result->massBinCenter()*0.001, intensity1);
+			g1->SetPointError(i, _binWidth*0.5, intensityErr1);
 
-		g2->SetPoint(i, result->massBinCenter()*0.001, intensity2);
-		g2->SetPointError(i, _binWidth*0.5, intensityErr2);
-
-		// check if we should make a transformation by 2pi
-		// this is needed because of cyclical variable phi
-		if(i>11){
-			double mpre;
-			double phpre;
-			gph->GetPoint(i-1,mpre,phpre);
-			double diff1=fabs(ph-phpre);
-			double diff2=fabs(ph+360-phpre);
-			double diff3=fabs(ph-360-phpre);
-			if(diff2<diff1 && diff2<diff3)ph+=360;
-			else if(diff3<diff1 && diff3<diff2)ph-=360;
-		}
+			if((numeric_limits<double>::has_infinity and intensity2 == numeric_limits<double>::infinity()) or intensity2!=intensity2) {
+				continue;
+			}
 
-		gph->SetPoint(i, result->massBinCenter()*0.001, ph);
-		gph->SetPointError(i, _binWidth*0.5, phErr);
+			g2->SetPoint(i, result->massBinCenter()*0.001, intensity2);
+			g2->SetPointError(i, _binWidth*0.5, intensityErr2);
+
+			// check if we should make a transformation by 2pi
+			// this is needed because of cyclical variable phi
+			if(i>11){
```

Comment:
No idea why that is there. From the looks of it, it probably is one of the first attempts to treat jumping phases. As the phases are now plotted +- 360 degrees anyway, I removed this.